Gifski/util.swift

```diff
@@ -2417,3 +2418,45 @@ extension URL {
 		try attributes.set("com.apple.metadata:\(key.attributeKey)", value: value)
 	}
 }
+
+extension NSViewController {
+	func push(viewController: NSViewController, completion: (() -> Void)? = nil) {
+		guard let window = view.window else {
+			return
+		}
+
+		let newOrigin = CGPoint(x: window.frame.midX - viewController.view.frame.width / 2.0, y: window.frame.midY - viewController.view.frame.height / 2.0)
+		let newWindowFrame = CGRect(origin: newOrigin, size: viewController.view.frame.size)
+
+		let animation = CABasicAnimation()
+		animation.toValue = newWindowFrame
+		animation.duration = 1.0
+
+		viewController.view.alphaValue = 0.0
+		NSAnimationContext.runAnimationGroup({ _ in
+			window.contentViewController?.view.animator().alphaValue = 0.0
+			window.contentViewController = nil
+			window.animator().setFrame(newWindowFrame, display: true)
+		}, completionHandler: {
+			window.contentViewController = viewController
+			viewController.view.animator().alphaValue = 1.0
+			window.makeFirstResponder(viewController)
+			completion?()
+		})
+	}
+
+	func add(childController: NSViewController) {
+		addChild(childController)
+		view.addSubview(childController.view)
+		childController.view.constrainEdgesToSuperview()
+	}
+}
+
+extension URL {
+	static func generateTempGifUrl(for videoUrl: URL) -> URL {
```

Comment:
I don't think this should be an extension on URL. It's too specific to this use-case.